src/db/couchDevices.ts

```diff
@@ -63,10 +77,22 @@ const loginIdDesign = makeJsDesign('loginId', ({ emit }) => ({
   }
 }))
 
+/**
+ * Looks up devices by the IP location.
+ */
+const locationDesign = makeJsDesign('location', ({ emit }) => ({
+  map: function (doc) {
+    if (doc.location == null || doc.location.error != null) return
+    emit([doc.location.country, doc.location.city, doc.location.region], null)
```

Comment:
This order will make it impossible to send notifications to just a state. We want the rows sorted like this:

```
...
USA, AZ, Phoenix
USA, CA, Escondido
USA, CA, San Diego
USA, CA, San Diego
USA, CA, San Francisco
USA, TX, San Marcos
...
Vatican, "", Vatican City
```

That way, we can use a key of `[USA, CA]` to target the whole state, or a key of `[USA, CA, San Diego]` to target just the city. With the sort you have now, there is no way to target the state:


```
...
USA, Escondido, CA
USA, Phoenix, AZ
USA, San Diego, CA
USA, San Diego, CA
USA, San Marcos, TX
USA, San Francisco, CA
...
Vatican, Vatican City, ""
```

See how Texas and Arizona are mixed in with the California entries now, making the index useless for getting the state?

It's nice that the "don't care" row is at the end for countries without states, but we can solve that by just defaulting the state to "" when we send notifications, which will match by default for countries where it doesn't matter.

Reply:
I see how your example is a problem where states get mixed up with the way I have the view structured.

The issue is that a lot of countries were we wouldn't perhaps care about the region actually do have regions. For example, "Paris, France" is actually in the region Île-de-France region (abbr. IDF). I figured that it'd be easier on marketing to not require that knowing the regions for those countries, which is why the last element is the region (the "don't care" element). Making the change as you suggested means we couldn't actually query for "Paris, France" without specifying its region, IDF.